src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -1128,6 +1128,14 @@ internal IEnumerable<PackageManagerSearchElementViewModel> Search(string searchT
                 string searchTerm = searchText.Trim();
                 var packages = new List<PackageManagerSearchElementViewModel>();
 
+                //The DirectoryReader and IndexSearcher have to be assigned after commiting indexing changes and before executing the Searcher.Search() method,otherwise new indexed info won't be reflected
+                LuceneSearchUtility.dirReader = LuceneSearchUtility.writer?.GetReader(applyAllDeletes: true);
+
+                if (LuceneSearchUtility.Searcher == null && LuceneSearchUtility.dirReader != null)
+                {
+                    LuceneSearchUtility.Searcher = new IndexSearcher(LuceneSearchUtility.dirReader);
+                }
```

Comment:
Placing this logic inside search() function is a bit strange, is there a better place in your opinion?

Reply:
We are setting the Searcher property similarly for Node search and NodeAutocomplete search also. https://github.com/DynamoDS/Dynamo/blob/2658b00f970f487cd83926ced349e08ad5469e77/src/DynamoCore/Search/NodeSearchModel.cs#L253. We want to set this only when searcher is null. What do you think?